builds/checkin/watchdog.yaml

```diff
@@ -0,0 +1,56 @@
+trigger: none
+pr:
+  branches:
+    include:
+      - master
+      - release/*
+      - iiot
+jobs:
+
+  ################################################################################
+  - job: check_run_pipeline
+  ################################################################################
+    displayName: Check pipeline preconditions (changes ARE in builds or edge hub)
+    pool:
+      vmImage: "ubuntu-16.04"
+    steps:
+      - bash: |
+          git log -m -1 --name-only --first-parent --pretty="" | egrep -i '^(builds|edge-hub/watchdog)'
+          if [[ $? == 0 ]]; then
+            echo "Detected changes inside builds or edgehub folders"
+            echo "##vso[task.setvariable variable=RUN_PIPELINE;isOutput=true]TRUE"
+          fi
+        displayName: Check changes in sources
+        name: check_files
+
+  ################################################################################
+  - job: linux_amd64
+  ################################################################################
+    displayName: Linux amd64
+    dependsOn: check_run_pipeline
+    condition: eq(dependencies.check_run_pipeline.outputs['check_files.RUN_PIPELINE'], 'true')
+    pool:
+      vmImage: "ubuntu-16.04"
+    steps:
+      - script: echo "##vso[task.setvariable variable=RUST_BACKTRACE;]1"
+        displayName: Set env variables
+      - bash: scripts/linux/generic-rust/install.sh --project-root "edge-hub/watchdog"
+        displayName: Install Rust
+      - bash: scripts/linux/generic-rust/build.sh --project-root "edge-hub/watchdog" --packages "watchdog"
+        displayName: Build
+
+  ################################################################################
+  - job: style_check
+  ################################################################################
+    displayName: Style Check
+    dependsOn: check_run_pipeline
+    condition: eq(dependencies.check_run_pipeline.outputs['check_files.RUN_PIPELINE'], 'true')
+    pool:
+      vmImage: "ubuntu-16.04"
+    steps:
+      - bash: scripts/linux/generic-rust/install.sh --project-root "edge-hub/watchdog"
+        displayName: Install Rust
+      - bash: scripts/linux/generic-rust/format.sh --project-root "edge-hub/watchdog"
```

Comment:
reminder: delete other format / install scripts when we have verified new ones work